utils.js

```diff
@@ -982,18 +978,80 @@ const parseGosumData = async function (gosumData) {
           name: name,
         });
       }
-      pkgList.push({
+      pkg = {
         group: group,
         name: name,
-        version: version,
-        _integrity: hash,
+        version: sumVersion,
+        _integrity: sumHash,
         license: license,
-      });
+      };
     }
   }
-  return pkgList;
+  return pkg;
+};
+exports.getGoPkgComponent = getGoPkgComponent;
+
+const parseGoModData = async function (goModData, goSumData) {
+  const pkgComponentsList = [];
+  let isModReplacement  = new Boolean(false);
+
+  if (!goModData) {
+    return pkgComponentsList;
+  }
+
+  const pkgs = goModData.split("\n");
+  for (let i in pkgs) {
+    const l = pkgs[i];
+
+    // Skip go.mod file headers, whitespace, and/or comments
+    if (l.includes("module ") || l.includes("go ")  || l.includes(")") || l.trim() === "" || l.trim().startsWith("//")){
+      continue;
+    }
+
+    // Handle required modules separately from replacement modules to ensure accuracy when parsing component data.
+    if (l.includes("require (")) {
+      isModReplacement = false;
+      continue;
+    } else if (l.includes("replace (")) {
+      isModReplacement = true;
+      continue;
+    }
+
+    const tmpA = l.trim().split(" ");
+
+    if (!isModReplacement) {
+      // Add group, name and version component properties for required modules
+      let group = path.dirname(tmpA[0]);
+      const name = path.basename(tmpA[0]);
+      if (group === ".") {
+        group = name;
+      }
+      const version = tmpA[1]
+      await getGoPkgComponent(goSumData, group, name, version)
```

Comment:
If you want you could `const component = await getGoPkgComponent(goSumData, group, name, version)` then add it to the array

Reply:
I went back and forth. IMO this is a more idiomatic promise handling paradigm. I am wondering if the upstream reviewers will approve of this